src/caffeine/allocation_m.f90

```diff
@@ -29,7 +29,6 @@ module allocation_m
     end type
 
     type :: prif_coarray_handle
-      private
```

Comment:
What are the implications of this change? 

I think we should continue to leverage the Fortran type system to strongly discourage PRIF clients from touching the unspecified metadata inside the `prif_coarray_handle`, because we want unfettered flexibility to change things below the PRIF interface. IMO this enforced encapsulation in our library implementation is far more important than enabling open-box testing, which seems to be the point in this commit.

IMHO we should revert this change, and either remove the test or deploy some test-specific workaround for inspecting library-internal state (notably using types and functions that are NOT `prif_`).

Reply:
@bonachea, Dan, I understand your preference. However, it is not just for the tests. I needed to remove that attribute to be able to pass the coarray address to C from `prif_base_pointer` as seen [here](https://github.com/BerkeleyLab/caffeine/pull/58/files#diff-228007d71dbe1e7e0f3f18a32feda012d02b2b9a1c21963b3b036b4e39ec3887R29). Also, Brad's @everythingfunctional [PR](https://github.com/BerkeleyLab/caffeine/pull/57/files#diff-f003feb27e90d53b50e3d227800f63d9e3e19d1014a51c8a19de524841dcc11fL32) also made this change when beginning to work on adding team functionalities, so it was a change I thought was necessary in the long run. Perhaps we should explore this topic more in our meeting this week to see if it is feasible for us to keep it `private`.

Reply:
I agree we should discuss further. 

I suspect the problem is the module structure of caffeine: it probably makes more sense for things like `module allocation_m` to instead be a `submodule` of `module prif`, and that "library-wide internals" like `prif_coarray_handle` and `handle_data` should instead be defined in `module prif` with appropriate `private` qualifiers.

Reply:
Yeah, I made it not private because the implementations in other modules needed access to the data, not because the tests did. I think all the coarray accesses/queries (i.e. allocation, RMA, image_index, etc) should go into the module where `prif_coarray_handle` is defined, but we can discuss in more detail tomorrow.

Reply:
I agree that it seems like we should update the module structure so that we can keep the component `private`. I am happy to do it in a future PR after we discuss it further tomorrow and agree on a plan.

Reply:
Added as issue #64

Reply:
As discussed in our meeting and tracked in issue #64, this restructuring to keep components `private` will occur in a future PR.